tools/binlog-slave-client.md

```diff
@@ -0,0 +1,149 @@
+  // data after modification
+  required Row row = 2;
+  // data before modification. It only takes effect for `Update MutationType`.
+  optional Row change_row = 3;
+}
+
+// `DMLData` stores all the mutations caused by DML in a table.
+message DMLData {
+  // `tables` contains all the table changes.
+  repeated Table tables = 1;
+}
+
+// `DDLData` stores the DDL information.
+message DDLData {
+  // the database used currently
+  optional string schema_name = 1;
+  // the relates table
+  optional string table_name = 2;
+  // `ddl_query` is the original DDL statement query.
+  optional bytes ddl_query = 3;
+}
+
+// `BinlogType` indicates the binlog type, including DML and DDL.
+enum BinlogType {
+  DML = 0; //  Has `dml_data`
+  DDL = 1; //  Has `ddl_query`
+}
+
+// `Binlog` stores all the changes in a transaction. Kafka stores the serialized result of the structure data.
+message Binlog {
+  optional BinlogType type = 1 [ (gogoproto.nullable) = false ];
+  optional int64 commit_ts = 2 [ (gogoproto.nullable) = false ];
+  // `dml_data` is marshalled from the DML type.
+  optional DMLData dml_data = 3;
+  optional DDLData ddl_data = 4;
+}
+```
+
+For the definition of the data format, see [`binlog.proto`](https://github.com/pingcap/tidb-tools/blob/master/tidb_binlog/slave_binlog_proto/proto/binlog.proto).
+
+### Driver
+
+TiDB-Tools provides Driver which is used to read the binlog data in Kafka. It has the following features:
+
+* Read the Kafka data.
+* Locate the binlog stored in Kafka based on `commit ts`.
+
+You need to configure the following information when using Driver:
+
+* `KafkaAddr`: the address of the Kafka cluster
+* `CommitTS`: from which `commit ts` to start reading the binlog
+* `Offset`: from which Kafka `offset` to start reading data. If `CommitTS` is set, you needn't configure this parameter
+* `ClusterID`: the cluster ID of the TiDB cluster
+
+You can use Driver by quoting the Driver code in package and refer to the example code provided by Driver to learn how to use Driver and parse the binlog data. 
+
+Currently, there are two examples:
+
+* Using Driver to synchronize data to MySQL. This example shows how to convert a binlog to SQL
+* Using Driver to print data
+
+The Driver address: [Binlog Slave Driver](https://github.com/pingcap/tidb-tools/tree/master/tidb_binlog/driver)。
+
+> **Notes:**
+>
+> - The example code only shows how to use Driver. If you want to use Driver in the production environment, you need to optimize the code.
+> - Currently, only Driver for golang and the example code are supported. If you want to use other languages, you need to generate the code file in the corresponding language based on the binlog proto file and develop an application to read the binlog data in Kafka, parse the data, and output the data to the downstream. You are also welcome to optimize the example code and submit the example code of other languages to [TiDB-Tools](https://github.com/pingcap/tidb-tools).
```

Comment:
The first sentence here is a little bit inappropriate. We can put it this way:

-> Currently, **only the Golang version of Driver and example code are available.**